keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -651,13 +610,119 @@ endpoints:
 
 The `ServiceImport.Spec.IP` (VIP) can be used to access this service from within this cluster. 
 
+### Supercluster Service Behavior Expectations
+
+#### Service Types
+
+- `ClusterIP`: This is the the straightforward case most of the proposal
+  assumes. Each `EndpointSlice` associated with the exported service is combined
+  with slices from other clusters to make up the supercluster service. They will
+  be imported to the cluster behind the supercluster IP.
+- `ClusterIP: none` (Headless): Headless services are supported and will be
+  imported with a `ServiceImport` and `EndpointSlices` like any other
+  `ClusterIP` service, but the way that they are exposed depends on the
+  cluster-level services from which they are derived. A supercluster service
+  will be exposed with a regular supercluster IP if any source cluster-level
+  service specifies a cluster IP. A supercluster service will be headless if and
+  only if all cluster-level services from which it is derived are headless.
```

Comment:
I'm sure you thought about it, but I just think it has to be pointed out explicitly.

Imagine the scenario:
- T1:  expose ClusterIP service from cluster A
- T2: expose headless service from cluster B
- T3: expose headless service from cluster C
- T4; stop exposing that ClusterIP service from cluster A

In that case, up until T4, the supercluster service will have  supercluster IP. At T4, all services are headless, so that supercluster IP should be removed. And in theory break someone?

@JeremyOT - thoughts?

Reply:
Is there a scenario where you'd ever actually want to change a given service from non-headless -> headless (or the reverse) and it wasn't a deployment error to begin with?

The thinking here is that if there's a mix of headless & non-headless services exported with the same name, some mistake has been made. We aim to surface that state with a condition on the `ServiceExport` as soon as the disagreement is detected. Failing to SuperclusterIP seems less risky as all clients (regardless of e.g. potentially poor DNS implementations) can consume a single VIP, but only real headless service consumers can be assumed to properly handle multiple/mutable IPs returned for the service.

I'll add clarification around the scenario you bring up.

Reply:
I didn't come up with any scenario that wouldn't be deployment error. And as a result I think that's probably fine.

But I would like to make it explicit so that people are aware of it.

Reply:
Service API tries to make it hard to change ClusterIP from a value to "none", but you can delete the Service and re-create it.

Reply:
called out this scenario